trin-core/src/portalnet/overlay.rs

```diff
@@ -563,4 +740,25 @@ mod test {
             Err(err) => assert!(err.to_string().contains(&msg)),
         }
     }
+
+    #[test]
```

Comment:
*nit* This test seems like it could be cleaned up a bit by using `rstest` like we do [here](https://github.com/ethereum/trin/blob/37c1931a2f2eb9e4832d980574070e8ff1a5311e/trin-core/src/utils/bootnodes.rs#L53).